spec/controllers/users_controller_spec.rb

```diff
@@ -16,97 +19,96 @@
       end
     end
-
-    context 'user was not found' do
-      before { get :show, params: { id: 10_000 } }
-
-      it do
-        expect(response).to redirect_to(users_path)
-        expect(flash[:danger]).to eq('User not found')
-      end
-    end
   end
 
   describe '[GET] #new' do
-    before { get :new }
+    before do
+      sign_in user
+      get :new
+    end
 
     it 'renders new user template' do
       expect(response).to render_template('new')
     end
   end
 
-  describe '[POST] #create' do
-    let(:params) { { first_name: user.first_name, surname: user.surname, age: user.age } }
-    context 'when params are valid' do
-      before do
-        post :create, params: { user: params }
-      end
-
-      it do
-        expect(User.find_by(*params[:user], id: user.id)).not_to be nil
-        expect(flash[:success]).to eq('New users successfully created')
-      end
-    end
-
-    context 'when params are not valid' do
-      before do
-        post :create, params: {
-          user: { first_name: nil, surname: user.surname, age: user.age }
-        }
-      end
-
-      it do
-        expect(flash[:danger]).to eq('User creation failed')
-        expect(response).to render_template('new')
-      end
-    end
-  end
+  # TODO: Refactor Test for user creation after pundit implementation
+  # describe '[POST] #create' do
+  #   let(:params) do
+  #     { first_name: user.first_name, surname: user.surname, age: user.age, email: user.email, password: user.password }
+  #   end
+
+  #   context 'when params are valid' do
+  #     before do
+  #       sign_in user
+  #       post :create, params: { user: params }
+  #     end
+
+  #     it do
+  #       expect(User.find_by(**params[:user], id: user.id)).not_to be nil
+  #       expect(flash[:success]).to eq('New users successfully created')
+  #     end
+  #   end
+
+  #   context 'when params are not valid' do
+  #     before do
+  #       sign_in user
+  #       params[:first_name] = nil
+  #       post :create, params: { user: params }
+  #     end
+
+  #     it do
+  #       expect(flash[:danger]).to eq('User creation failed')
+  #       expect(response).to render_template('new')
+  #     end
+  #   end
+  # end
 
   describe '[GET] #edit' do
     context 'when user is found' do
-      before { get :edit, params: { id: user.id } }
+      before do
+        sign_in user
+        get :edit, params: { id: user.id }
+      end
 
       it do
         expect(response).to render_template('edit')
         expect(assigns(:user)).to eq(user)
       end
     end
+  end
 
-    context 'when user is not found' do
-      before { get :edit, params: { id: 10_000 } }
+  describe '[PUT] #update' do
+    let(:params) do
+      { id: user.id, user: { first_name: 'NewFirstName', surname: 'NewSurname', age: 7 } }
+    end
 
-      it do
-        expect(flash[:danger]).to include('User not found')
-      end
+    before do
+      sign_in user
+      put :update, params: params
     end
```

Comment:
Why are you deleting the test?

Reply:
Because the user can only access the current user information, regardless of the id that appears in the url (This will be changed when I implement pundit)

Reply:
OK, in this case, add a TODO comment to the test